components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AccountConfirmationValidationHandler.java

```diff
@@ -88,7 +92,9 @@ public void handleEvent(Event event) throws IdentityEventException {
 
         if (IdentityEventConstants.Event.POST_AUTHENTICATION.equals(event.getEventName())) {
             if (log.isDebugEnabled()) {
-                log.debug("Handling PostAuthenticate for " + user);
+                log.debug(
+                        String.format("Handling PostAuthenticate for user: %s in userstore: %s in tenant: %s",
+                                Utils.maskIfRequired(userName), domainName, tenantDomain));
```

Comment:
Previously,  in this debug log, it logged user info in the following manner, exposing their username, userstore domain and tenant domain (This was due to the `toString()` [1] implementation of the User class). 
- Handling PostAuthenticate for DEFAULT/jane@gmail.com@testasgardeo1

But this exposes PII with their username. Therefore, it is improved to log the same information (with masked username) in debug mode in a different format as follows as it is done in other places [2].
-  Handling PostAuthenticate for user: j************m in userstore: DEFAULT in tenant: testasgardeo1

[1] - https://github.com/wso2/carbon-identity-framework/blob/e8f72f4b58c61e70c88df467dfd784f9e9f161ae/components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/User.java#L298-L312
[2] - https://github.com/wso2-extensions/identity-governance/blob/79adb68175abb5a25db779c3ad011d221a995608/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AccountConfirmationValidationHandler.java#L121-L122

Reply:
Since this is a debug log, we don't need to mask the PII.